src/SFML/Window/Unix/KeyboardImpl.cpp

```diff
@@ -0,0 +1,586 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2018 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/Unix/KeyboardImpl.hpp>
+#include <SFML/Window/Unix/Display.hpp>
+#include <SFML/Window/Unix/KeySymToKeyMapping.hpp>
+#include <SFML/Window/Unix/KeySymToUnicodeMapping.hpp>
+#include <SFML/System/String.hpp>
+#include <SFML/System/Utf.hpp>
+#include <X11/keysym.h>
+#include <X11/Xlib.h>
+#include <X11/XKBlib.h>
+#include <cstring> // strcmp
+
+namespace
+{
+
+const KeyCode NullKeyCode = 0;
+KeyCode scancodeToKeycode[sf::Keyboard::ScanCodeCount];     ///< Mapping of SFML scancode to X11 KeyCode
```

Comment:
Is the space before the comment meant for alignment with the line below? I can't tell what is the intention here.